### PR TITLE
Make showTileBoundaries prefer a vector source

### LINKED ISSUE #833: showTileBoundaries fails to ignore raster sources as documented
When I have a raster layer, `map.showTileBoundaries = true` is showing 0kb:

Screenshot 2022-01-23 17 42 05

But it is documented as:

> The uncompressed file size of the first vector source is drawn in the top left corner of each tile, next to the tile ID.

Which means that it should be skipping the raster layer and showing the size of my vector source.

When I hide the raster layer with `map.setLayoutProperty('simple-tiles','visibility','none');`, it works correctly:

Screenshot 2022-01-23 17 42 17

Note that the raster tiles were 256px, while the vector tiles are 512px.

Also, `kb` should be `kB`.

Runnable example: https://jsbin.com/fohaxat

### PR DESCRIPTION
**Pull request by jleedev**
Fixes #833

### CODE REVIEW

**Comment by jleedev** [comment]
I don't know why the render test doesn't trip. Manual test:

Screenshot 2022-07-26 07 24 36

**Comment by HarelM** [comment]
Can you please add tests and update the changelog.md file?

**Comment by HarelM** [comment]
Any change for a unit test?

**Comment by jleedev** [comment]
Could factor out the logic to an exported function in either painter.ts or draw_debug.ts, then invoke it with various maps and assert that it returns the desired SourceCache.

**Comment by HarelM** [comment]
Exporting the show tile boundaries to a different method/file and testing it would be great, thanks!

**Comment by HarelM** [comment]
Any updates on this?

**Comment by jleedev** [comment]
I think I can find some time this week.

**Comment by HarelM** [comment]
Any updates on this? I would really like to push this forward...

**Comment by HarelM** [comment]
@jleedev any news?

File: src/render/painter.ts [hunk 1]
```diff
@@ -483,20 +483,7 @@ class Painter {
...
-                if (layer.source && !layer.isHidden(this.transform.zoom)) {
-                    if (layer.source !== (sourceCache && sourceCache.id)) {
-                        sourceCache = this.style.sourceCaches[layer.source];
-                    }
-                    if (!selectedSource || (selectedSource.getSource().maxzoom < sourceCache.getSource().maxzoom)) {
-                        selectedSource = sourceCache;
-                    }
-                }
-            });
+            const selectedSource = selectDebugSource(this);
```
Comment (HarelM) [inline src/render/painter.ts:486]: Looks a lot better! Thanks! Can we pass something more specific here?
Whenever I see "this" as a parameter I get itchy...

Comment (jleedev) [reply src/render/painter.ts:486]: Sure, looks like it just needs the Style and the current zoom number.

**Review by HarelM — APPROVED** [review]

**Comment by jleedev** [comment]
Thanks for the code review.

**Pull request merged**